packages/evm/test/court/controller/controller-modules.js

```diff
@@ -616,4 +617,118 @@ contract('Controller', ([_, fundsGovernor, configGovernor, modulesGovernor, some
+      module = await ControlledMock.new(controller.address)
+    })
+
+    context('when the sender is the governor', () => {
+      const from = modulesGovernor
+
+      context('when setting a function', () => {
+        const itRegistersTheCustomFunction = () => {
+          it('sets the function', async () => {
+            const receipt = await controller.setCustomFunction(setCounterSig, module.address, { from })
+            assertAmountOfEvents(receipt, CONTROLLER_EVENTS.CUSTOM_FUNCTION_SET)
+            assertEvent(receipt, CONTROLLER_EVENTS.CUSTOM_FUNCTION_SET, { signature: setCounterSig, target: module.address })
+
+            const anotherReceipt = await controller.setCustomFunction(receiveEtherSig, module.address, { from })
+            assertAmountOfEvents(anotherReceipt, CONTROLLER_EVENTS.CUSTOM_FUNCTION_SET)
+            assertEvent(anotherReceipt, CONTROLLER_EVENTS.CUSTOM_FUNCTION_SET, { signature: setCounterSig, target: module.address })
+          })
+
+          it('can be called', async () => {
+            const data = module.contract.methods.setCounter(10).encodeABI()
+            await controller.setCustomFunction(setCounterSig, module.address, { from })
+
+            assertBn(await module.counter(), 0, 'counter does not match')
+            await controller.sendTransaction({ data })
+            assertBn(await module.counter(), 10, 'counter does not match')
+          })
+
+          it('handles eth transfers properly', async () => {
+            await controller.setCustomFunction(receiveEtherSig, module.address, { from })
+
+            const receipt = await controller.sendTransaction({ data: receiveEtherSig, value: bigExp(1, 18) })
+
+            assertAmountOfEvents(receipt, 'EtherReceived', { decodeForAbi: ControlledMock.abi })
+            assertEvent(receipt, 'EtherReceived', { expectedArgs: { sender: controller.address, value: bigExp(1, 18) }, decodeForAbi: ControlledMock.abi })
+
+            const currentBalance = await web3.eth.getBalance(module.address)
+            assertBn(currentBalance, bigExp(1, 18), 'module balance does not match')
+          })
+
+          it('handles reverts properly', async () => {
+            await controller.setCustomFunction(failSig, module.address, { from })
+
+            await assertRevert(controller.sendTransaction({ data: failSig }), 'CONTROLLED_FAIL')
+          })
+        }
+
+        context('when the function was not set', () => {
+          itRegistersTheCustomFunction()
+        })
+
+        context('when the function was already set', () => {
+          beforeEach('set custom function', async () => {
+            await controller.setCustomFunction(failSig, modulesGovernor, { from })
+            await controller.setCustomFunction(setCounterSig, modulesGovernor, { from })
+            await controller.setCustomFunction(receiveEtherSig, modulesGovernor, { from })
+          })
+
+          itRegistersTheCustomFunction()
+        })
```

Comment:
Perhaps another test we should add is an attempt at overriding an existing function signature.

Reply:
Addressed in 3e5d18a6eeb5c9e20f57eacea79ba35151820a98